runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -1325,6 +1325,17 @@ def DeleteCommands()
   sign_undefine(BreakpointSigns->map("'debugBreakpoint' .. v:val"))
 enddef
 
+def QuoteArg(x: string): string
+  var x_escaped = x ->substitute('\\', '\\\\', 'g')
+    ->substitute('"', '\\"', 'g')
+
+  var x_list = split(x_escaped, '\zs')
+  if x_list[0] == '"' && x_list[-1] == '"'
+    return x_escaped
+  else
+    return $"\"{x_escaped}\""
+  endif
```

Comment:
I guess this is supposed to quote the string unless it is already quoted. According to the gdb/mi spec, the command parameter `x` does not have **any** `"` characters.  I think the use of `$"..."` is wrong because it interprets escape characters, `$'...'` is better, or even simply `'"' .. x_escaped .. '"'. In any event, I think something like the following can replace all that `split()` ... stuff.
```
return x_escaped[0] == '"' ? x_escaped : $'"{x_escaped}"'
```

Reply:
> According to the gdb/mi spec, the command parameter x does not have any " characters

You keep repeating that but I don't see any evidence in the docs. What's stated is that the string may be either unquoted or a C-string with double quotes, meaning that you can have escape sequences embedded (see [here](https://github.com/bminor/binutils-gdb/blob/a81f4e591fdb8530d832addc39beb31353b0ef2d/gdb/mi/mi-parse.c#L38)) and quotes (that, according to the source, can be escaped using backslashes.

Reply:
I was not aware that in Vim you can index a string like a list. Though, it does not seem to work for negative indices (e.g. `x_escaped[-1]`). Is it enough to check check only the value at index `0`? Or shall I also check if the last character is a string as well? Because in the latter case I think the `split` would be needed and the following should be correct:

```
  var x_list = split(x_escaped, '\zs')
  return  if x_list[0] == '"' && x_list[-1] == '"' ? x_escaped : $'"{x_escaped}"'
```

Reply:
you would need to use `x_escaped[-1:]`. There are some subtle differences between legacy Vim script and Vim9 script, but I think the new Vim9 script behaviour is more reasonable (you will always get characters, not bytes, which matters for multi-byte characters).

Reply:
> > According to the gdb/mi spec, the command parameter x does not have any " characters
> 
> You keep repeating that but I don't see any evidence in the docs. What's stated is that the string may be either unquoted or a C-string with double quotes, meaning that you can have escape sequences embedded (see [here](https://github.com/bminor/binutils-gdb/blob/a81f4e591fdb8530d832addc39beb31353b0ef2d/gdb/mi/mi-parse.c#L38)) and quotes (that, according to the source, can be escaped using backslashes.

@LemonBoy Certainly the code is the ultimate spec. https://github.com/vim/vim/pull/15261#issuecomment-2229399949 is copied from the gdb/mi docs. Am I misunderstanding `c-string`?

Reply:
> Am I misunderstanding `c-string`?

Doh! This is a normal "C" quoted string? I think I read `non-blank-sequence` and spun out from that.

Reply:
> you would need to use `x_escaped[-1:]`

?
```
vim9script
var x = "abc"
echo x[0]
echo x[-1]
echo x[-1 : ]
```
outputs
```
a
c
c
```

Reply:
well, seems another subtle difference between Vim9 script and legacy script.